File: go/linked-list-random-node.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

type Solution struct {
	NodeValues []int
}

func Constructor(head *ListNode) Solution {
	current := head
	nodeValues := make([]int, 0)
	for current != nil {
		nodeValues = append(nodeValues, current.Val)
		current = current.Next
	}
	return Solution{NodeValues: nodeValues}
}

func (this *Solution) GetRandom() int {
	i := rand.Intn(len(this.NodeValues))
	return this.NodeValues[i]
}

/**
 * Your Solution object will be instantiated and called as such:
 * obj := Constructor(head);
 * param_1 := obj.GetRandom();
 */

File: python/linked-list-random-node.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def __init__(self, head: Optional[ListNode]):
        self.nodes = []
        current = head
        while current:
            self.nodes.append(current.val)
            current = current.next

    def getRandom(self) -> int:
        val = self.nodes[randint(0, len(self.nodes) - 1)]
        return val

# Your Solution object will be instantiated and called as such:
# obj = Solution(head)
# param_1 = obj.getRandom()
